aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -191,180 +254,261 @@ impl Execution {
+                    .map(|res| res[0].clone().into()),
+            )
+        } else {
+            Some(
+                executor
+                    .execute_transaction_block(vec![txn.clone()])
+                    .map(|res| res[0].clone().into()),
+            )
+        }
     }
 
     fn print_mismatches(
+        &self,
         cur_version: u64,
         res_1: &Result<(WriteSet, Vec<ContractEvent>), VMStatus>,
         res_2: &Result<(WriteSet, Vec<ContractEvent>), VMStatus>,
     ) {
         match (res_1, res_2) {
             (Err(e1), Err(e2)) => {
-                if e1 != e2 {
-                    println!("error is different at {}", cur_version);
-                    println!("error {} is raised from V1", e1);
-                    println!("error {} is raised from V2", e2);
+                if e1.message() != e2.message() || e1.status_code() != e2.status_code() {
+                    self.output_result_str(format!(
+                        "error is different at version:{}",
+                        cur_version
+                    ));
+                    // self.output_result_str(format!("error {} is raised from V1", e1));
+                    // self.output_result_str(format!("error {} is raised from V2", e2));
                 }
             },
-            (Err(e), Ok(res)) => {
-                println!("error {} is raised from V1 at {}", e, cur_version);
-                println!(
-                    "output from V2 at version:{}\nwrite set:{:?}\n events:{:?}\n",
-                    cur_version, res.0, res.1
-                );
+            (Err(_), Ok(_)) => {
+                self.output_result_str(format!(
+                    "V1 returns error while V2 does not at version:{}",
+                    cur_version
+                ));
+                // self.output_result_str(format!(
+                //     "output from V2 at version:{}\nwrite set:{:?}\n events:{:?}\n",
+                //     cur_version, res.0, res.1
+                // ));
             },
-            (Ok(res), Err(e)) => {
-                println!("error {} is raised from V2 at {}", e, cur_version);
-                println!(
-                    "output from V1 at version:{}\nwrite set:{:?}\n events:{:?}\n",
-                    cur_version, res.0, res.1
-                );
+            (Ok(_), Err(_)) => {
+                self.output_result_str(format!(
+                    "V2 returns error while V1 does not at version:{}",
+                    cur_version
+                ));
+                // self.output_result_str(format!(
+                //     "output from V1 at version:{}\nwrite set:{:?}\n events:{:?}\n",
+                //     cur_version, res.0, res.1
+                // ));
             },
             (Ok(res_1), Ok(res_2)) => {
                 // compare events
+                let mut error = false;
+                if res_1.1.len() != res_2.1.len() {
+                    error = true;
+                }
```

Comment:
This still panics if sizes are different? I think you want to have if != then error else do a detailed comparison? Same for write sets?

Reply:
done